src/presentation/http/router/note.test.ts

```diff
@@ -159,6 +159,155 @@ describe('Note API', () => {
       }
     });
 
+    test.each([
+      /** returns 200 and note with accessRights if user is in inherited team with role write */
+      {
+        roleInRootTeam: MemberRole.Write,
+        intermidiateTeamDefined: false,
+        expectedStatusCode: 200,
+      },
+      /** returns 403 and 'Permission denied' message if intermediate team without user was inherited */
+      {
+        roleInRootTeam: MemberRole.Write,
+        intermidiateTeamDefined: true,
+        roleInIntermidiateTeam: undefined,
+        expectedMessage: 'Permission denied',
+        expectedStatusCode: 403,
+      },
+      /** returns 200 and note with accessRights if user is in inherited team with role read */
+      {
+        roleInRootTeam: MemberRole.Read,
+        intermidiateTeamDefined: false,
+        expectedStatusCode: 200,
+      },
+      /** returns 403 and 'Permission denied' message if intermediate team without user was inherited */
+      {
+        roleInRootTeam: MemberRole.Read,
+        intermidiateTeamDefined: true,
+        roleInIntermidiateTeam: undefined,
+        expectedMessage: 'Permission denied',
+        expectedStatusCode: 403,
+      },
+      /** returns 403 and 'Permission denied' message if user is not in inherited team of the note */
+      {
+        roleInRootTeam: null,
+        intermidiateTeamDefined: false,
+        expectedMessage: 'Permission denied',
+        expectedStatusCode: 403,
+      },
+      /** returns 403 and 'Permission denied' message if intermediate team without user was inherited */
+      {
+        roleInRootTeam: null,
+        intermidiateTeamDefined: true,
+        roleInIntermidiateTeam: undefined,
```

Comment:
Why you use `null` in roleInRootTeam, but `undefined` in roleInIntermidiateTeam?